lib/galaxy/tool_util/data/__init__.py

```diff
@@ -250,6 +250,11 @@ def add_entries(
                 self.add_entry(
                     entry, allow_duplicates=allow_duplicates, persist=persist, entry_source=entry_source, **kwd
                 )
+            except MessageException as e:
+                if e.type == "warning":
+                    log.warning(str(e))
+                else:
+                    log.error(str(e))
```

Comment:
The default for `MessageException.type` is `info`, so we probably need a more complete helper method in `lib/galaxy/exceptions/__init__.py` that logs a `MessageException` at the right level.

Reply:
I think you would always want to control this in the calling context ?

Reply:
(since here we're calling error on the default log level)

Reply:
What I'm saying is that *if* the caught exception was raised as `MessageException("foo")` (i.e. `MessageException("foo", type="info")`), this would be logged as error.
I can open a PR to make it clearer, I suppose.